src/lib/arch/aarch64/Architecture.cc

```diff
@@ -126,23 +122,17 @@ Architecture::Architecture(kernel::Linux& kernel, ryml::ConstNodeRef config)
       ryml::ConstNodeRef opcode_node =
           config["Ports"][i]["Instruction-Opcode-Support"];
       for (size_t j = 0; j < opcode_node.num_children(); j++) {
-        // If latency information hasn't been defined, set to zero as to inform
-        // later access to use group defined latencies instead
+        // If latency information hasn't been defined, set to zero as to
+        // inform later access to use group defined latencies instead
         uint16_t opcode = opcode_node[j].as<uint16_t>();
-        opcodeExecutionInfo_.try_emplace(
-            opcode, simeng::arch::aarch64::ExecutionInfo{0, 0, {}});
+        opcodeExecutionInfo_.try_emplace(opcode, ExecutionInfo{0, 0, {}});
         opcodeExecutionInfo_[opcode].ports.push_back(static_cast<uint8_t>(i));
       }
     }
   }
 }
-Architecture::~Architecture() {
-  cs_close(&capstoneHandle);
-  decodeCache.clear();
-  metadataCache.clear();
-  groupExecutionInfo_.clear();
-  SVCRval_ = 0;
-}
+
+Architecture::~Architecture() { cs_close(&capstoneHandle_); }
```

Comment:
Why do we not need to clear the caches and groupExecutionInfo on deconstruction now?

Reply:
As they are no longer static, they will be destructed between instantiations of the Architecture class